### PR TITLE
Made VerifyAttestationInformation interface asynchronous

### PR DESCRIPTION
**Pull request by vijs**
#### Problem
* Device Attestation Verifier’s VerifyAttestationInformation method needs to be asynchronous (non-blocking) so that long running processes do not block the main SDK event loop.
* Reference: Issue #11917

#### Change overview
* Add callback parameter to VerifyAttestationInformation method.
* Updated DeviceController with new callback to execute once commissioner is done validating the attestation information.

#### Testing
* Matter Unit Tests

### CODE REVIEW

File: src/credentials/examples/DefaultDeviceAttestationVerifier.cpp
Comment (tcarmelveilleux) [inline src/credentials/examples/DefaultDeviceAttestationVerifier.cpp:224]: These are dual-side-effect macros. Very hard to understand. Please only use `attestationError` value to map to CHIP_NO_ERROR/CHIP_ERROR...

Since the result is async, don't expect to return errors to caller, only expect callback to have been called

**Review by woody-apple — APPROVED** [review]

File: src/credentials/examples/DefaultDeviceAttestationVerifier.cpp
Comment (vijs) [reply src/credentials/examples/DefaultDeviceAttestationVerifier.cpp]: Addressed in 1a5487c

**Comment by bzbarsky-apple** [comment]
This pull request seems to be making all sorts of unrelated changes and include all sorts of unrelated commits.  Please rebase on top of tip.

**Comment by vijs** [comment]
Waiting on approvals to merge this PR

**Review by andy31415 — APPROVED** [review]

**Comment by msandstedt** [comment]
Callback::Callback isn't type safe, and also has a deleted copy constructor that is an artifact that in some contexts allows Callbacks to be prematurely canceled after being enqueued.  But that facility isn't used here.

A delegate interface would have been a better choice.

**Pull request merged**
